src/scheduler.cpp

```diff
@@ -35,7 +35,7 @@ void Scheduler::threadMain()
 		}
 
 		// the mutex is locked again now...
-		if (ret == std::cv_status::timeout) {
+		if ((ret == std::cv_status::timeout) && !eventList.empty()) {
```

Comment:
You added redundant parenthesis.

Reply:
@ranisalt so you want me to remove the ()? i thought it was needed as there is two checks in the same IF statement.

Reply:
It is not. Logical expressions have lower priority than equality or unary expressions, so it would parse as expected.